File: app/src/main/java/com/example/james/myapplication/Model/ImageItem.java
package com.example.james.myapplication.Model;

import android.support.annotation.NonNull;

public class ImageItem implements Comparable<ImageItem> {
    private String name;
    private String rootPath;
    private String userPath;

    public String getSize() {
        return size;
    }

    private String size;

    public Boolean getSelected() {
        return isSelected;
    }

    public void setSelected(Boolean selected) {
        isSelected = selected;
    }

    public Boolean getMounted() {
        return isMounted;
    }

    public void setMounted(Boolean mounted) {
        isMounted = mounted;
    }

    private Boolean isSelected = false;
    private Boolean isMounted = false;

    public ImageItem(String name, String rootPath, String userPath, String size){
        this.name = name;
        this.rootPath = rootPath;
        this.userPath = userPath;
        this.size = size;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRootPath() {
        return rootPath;
    }

    public String getUserPath() {
        return userPath;
    }


    @Override
    public String toString() {
        return name;
    }

    @Override
    public int compareTo(@NonNull ImageItem o) {
        return this.name.compareTo(o.name);
    }
}

File: app/src/main/java/com/jimzrt/umsmounter/model/ImageItem.kt
package com.jimzrt.umsmounter.model

class ImageItem(var name: String, var rootPath: String, var userPath: String, var size: String) : Comparable<ImageItem> {
    var url: String? = null
    var isDownloading = false

    @JvmField
    var progress = 0
    var downloadId = -1
    var selected = false
    var mounted = false

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val imageItem = other as ImageItem
        return name == imageItem.name
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    override fun toString(): String {
        return name
    }

    override fun compareTo(other: ImageItem): Int {
        return name.compareTo(other.name)
    }

}